kiosk/src/Models/Kiosk.ts

```diff
@@ -37,10 +40,48 @@ export class Kiosk {
 
         try {
             this.games = (await response.json()).games;
+            this.games.push()
         }
         catch (error) {
             throw new Error(`Unable to process game list downloaded from "${url}": ${error}`);
         }
+
+        // the added games persist in local storage, but not the live game list
+        // that feeds the carousel. That's what this function is for.
+        this.addNewGamesToList();
+    }
+
+    saveNewGame(gameId: string): GameData | undefined {
+        const allAddedGames = this.getAllAddedGames();
+        if (!allAddedGames[gameId]) {
+            this.numAddedGames += 1;
+            const newGame = new GameData(gameId, `Kiosk Game ${this.numAddedGames}`, this.addedGameDescription, "None");
+            this.games.push(newGame);
+            allAddedGames[gameId] = newGame;
+            localStorage.setItem(this.addedGamesLocalStorageKey, JSON.stringify(allAddedGames));
+            return newGame;
+        }
+    }
+
+    getAllAddedGames(): { [index: string]: GameData } {
+        const json = localStorage.getItem(this.addedGamesLocalStorageKey);
+        if (!json) {
+            return {};
+        }
+        const allAddedGames: { [index: string]: GameData } = JSON.parse(json);
+        return allAddedGames;
+    }
+
+    addNewGamesToList() : void {
+        // check if there are custom games to add to list
+        const addedGames = localStorage.getItem(this.addedGamesLocalStorageKey);
+        if (addedGames) {
+            const addedGamesJson: Object = JSON.parse(addedGames);
```

Comment:
don't use the Object type. if possible specify the actual type (or use `any` as a last resort)